### ISSUE TITLE
How keypoint detection models are trained

### ISSUE THREAD

**Issue by trouble-maker007** [ivanpuhachov/line-drawing-vectorization-polyvector-flow#5]
Can you give an overview of how the model for keypoint detection is trained?

**Comment by ivanpuhachov** [comment]
Hi, we outlined training details in Appendix in the paper here [https://www-labs.iro.umontreal.ca/~bmpix/pdf/polyvector_flow.pdf](https://www-labs.iro.umontreal.ca/~bmpix/pdf/polyvector_flow.pdf)
I believe the most important part was to train on random crops with increasing resolution. Masked loss was crucial, otherwise, the model overfits to 0 prediction (no keypoints). Let me know if you have any specific questions.

**Comment by trouble-maker007** [comment]
@ivanpuhachov Thanks a lot for the detailed explanation!

Existing open source solutions are the key points of the human body, do you know the similar to the intersection of the lines or key point detection open source solutions? Will you follow up with open source code for this key point detection?

**Comment by ivanpuhachov** [comment]
Human body / face keypoints detectors typically have a fixed number of points to predict. This case is more similar to general object detection, but the bounding box is not defined. This is why we used CenterNet approach (essentially predicting heatmap over the image for possible intersections). 
For open-source detectors I would redirect you to detectron or mmdetection – they have a large community and thorough documentation, and it should be straightforward to adapt it to keypoint detection.

**Comment by trouble-maker007** [comment]
@ivanpuhachov Thank you very much for your detailed answer. I will try on my own dataset.

**Comment by ivanpuhachov** [comment]
Good luck. I'm closing the issue
